pandas/core/frame.py

```diff
@@ -10460,6 +10460,118 @@ def quantile(
         result = self._constructor(res)
         return result
 
+    def quantiles(
+        self,
+        q=0.5,
+        axis: Axis = 0,
+        numeric_only: bool = True,
+        interpolation: str = "nearest",
+    ):
+        """
+        Return values at the given quantile over requested axis for all columns.
```

Comment:
this should all be in the quantile method in algos not here

Reply:
For clarity, do you mean that this code should be in a new `quantile` method in algos that handles the table case, or in `BlockManager.quantile`, where it looks like the internal implementation of `DataFrame.quantile` resides?